README.md

```diff
@@ -1,85 +1,155 @@
+We used Docker to containerize our various micro-services which serves our combined application.
+
+The combined application contains qualities of:
+* Question Repository Application
+* User Profile Management Application
+
+We have various micro-services including:
+* MySQL Database
+* User Service
+* Auth Service
+* Question Service
+* Frontend Service (Web UI)
+
+Additional Notes:
+
+If you are not using Windows OS, and would like a equivalent instruction for a differnt OS, do reach out to us for clarification.
+
+**For clarifications, do leave your questions at [Feedback PR](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g32/pull/1) created in our repository.**
+
+# Requirements
+
+## Network
+
+Internet connection is required.
 
-- [NodeJS](https://nodejs.org/en/download)
-- [MySQL](https://dev.mysql.com/downloads/mysql/)
+**[ IMPORTANT! ]** Do _not_ test while on NUS grounds or connected
+(either directly or indirectly) to NUS Wifi.
+(NUS networks blocks MongoDB which is required by our application.) 
 
-For development, you may also want to install:
+## Software
+Download and install these software if you do not have them locally.
 
-- [MongoDB Compass](https://www.mongodb.com/try/download/compass)
 - [Docker Desktop](https://www.docker.com/get-started/)
-- [Postman](https://www.postman.com/downloads/)
 
-# Setup
+# Testing
 
-## Setup MySQL locally
+For clarity, the commands given below should be executed at the root directory if not specified otherwise.
 
-1. Search for **MySQL 8.1 Command Line Client** on your computer. Execute it to open up the terminal.
-2. Enter your root password.
-3. Run the SQL statements in `./User/schema.sql` on the terminal.
+_The commands below are provided for convenience and may be incorrect.
+If reading the textual instruction is not clear enough, do reach out to us for clarification._
 
-## Setup environment variables
+## Pre-testing Set up
 
-1. Enter the following command
-   ```
-   cp template.env .env
-   ```
-2. Open `.env` file
-3. Enter root password (previously configured when installing MySQL)  
-   Example: if your root password is "password1234",
-   `MY_SQL_PWD=password1234`
-4. Enter JWT token password (for generating and decoding JWT tokens)  
-   Example: if your root password is "password",
-   `JWT_SECRET_KEY=password`
+### Clone repository
 
-## Install NodeJS packages
+Clone the repository locally to your device (laptop/computer).
 
+```shell
+git clone https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g32.git
 ```
-npm i install-all && npm run install-all
+
+Note: If the above not work, please use the correct link or download the source code directly from the release.
+
+### Setup environment variables
+
+Duplicate `template.env` as `.env` at the root directory.
+
+```shell
+cp template.env .env
 ```
 
-# Start Application
+### Start Docker Daemon
 
-Start local MySQL Server (Windows Service), if necessary.
+For Windows, start _Docker Desktop_.
```

Comment:
macOS has Docker Desktop too, so i think we can include "For windows and macOS ..."

Reply:
Did you try to follow the steps in the README?

Reply:
all good on macOS

Reply:
all good on macOS